tokens.go

```diff
@@ -0,0 +1,228 @@
+	Scopes        []string `json:"scopes"`
+	ExpirySeconds *int     `json:"expiry_seconds,omitempty"`
+}
+
+// TokenUpdateRequest represents a request to update a token.
+type TokenUpdateRequest struct {
+	Name   string   `json:"name,omitempty"`
+	Scopes []string `json:"scopes,omitempty"`
+}
+
+// TokenScope is a representation of a scope for the public API.
+type TokenScope struct {
+	Name string `json:"name"`
+}
+
+type tokenScopesRoot struct {
+	TokenScopes []TokenScope `json:"scopes"`
+	Links       *Links       `json:"links"`
+	Meta        *Meta        `json:"meta"`
+}
+
+type tokenScopeNamespaceParam struct {
+	Namespace string `url:"namespace,omitempty"`
+}
+
+// List all DigitalOcean API access tokens.
+func (c TokensServiceOp) List(ctx context.Context, opt *ListOptions) ([]Token, *Response, error) {
+	path, err := addOptions(accessTokensBasePath, opt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := c.client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(tokensRoot)
+	resp, err := c.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+	if l := root.Links; l != nil {
+		resp.Links = l
+	}
+	if m := root.Meta; m != nil {
+		resp.Meta = m
+	}
+
+	return root.Tokens, resp, err
+}
+
+// Get a specific DigitalOcean API access token.
+func (c TokensServiceOp) Get(ctx context.Context, tokenID int) (*Token, *Response, error) {
+	path := fmt.Sprintf("%s/%d", accessTokensBasePath, tokenID)
+	req, err := c.client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(tokenRoot)
+	resp, err := c.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root.Token, resp, err
+}
+
+// Create a new DigitalOcean API access token.
+func (c TokensServiceOp) Create(ctx context.Context, createRequest *TokenCreateRequest) (*Token, *Response, error) {
+	req, err := c.client.NewRequest(ctx, http.MethodPost, accessTokensBasePath, createRequest)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(tokenRoot)
+	resp, err := c.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root.Token, resp, err
+}
+
+// Update the name or scopes of a specific DigitalOcean API access token.
+func (c TokensServiceOp) Update(ctx context.Context, tokenID int, updateRequest *TokenUpdateRequest) (*Token, *Response, error) {
+	path := fmt.Sprintf("%s/%d", accessTokensBasePath, tokenID)
+	req, err := c.client.NewRequest(ctx, http.MethodPatch, path, updateRequest)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(tokenRoot)
+	resp, err := c.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root.Token, resp, err
+}
+
+// Revoke a specific DigitalOcean API access token.
+func (c TokensServiceOp) Revoke(ctx context.Context, id int) (*Response, error) {
+	path := fmt.Sprintf("%s/%d", accessTokensBasePath, id)
```

Comment:
```suggestion
func (c TokensServiceOp) Revoke(ctx context.Context, tokenId int) (*Response, error) {
	path := fmt.Sprintf("%s/%d", accessTokensBasePath, tokenId)
```

we use `tokenID` in `GET` request. should we use the same for the `REVOKE` request?

Reply:
Good catch. Updated.